integrations/SnapPixel/browser.js

```diff
@@ -78,27 +75,22 @@ class SnapPixel {
 
     const cookieData = Storage.getUserTraits();
 
-    let payload = {
-      user_email: cookieData.email,
-      user_phone_number: cookieData.phone,
-    };
-
-    if (!payload.user_email && !payload.user_phone_number) {
-      logger.debug(
-        "User parameter (email or phone number) not found in cookie. identify is required"
-      );
-      return;
-    }
+    const userEmail = cookieData.email;
+    const userPhoneNumber = cookieData.phone;
 
+    let payload = {};
     if (this.hashMethod === "sha256") {
-      if (isDefinedAndNotNull(payload.user_email)) {
-        payload.user_email = sha256(payload.user_email).toString();
-      }
-      if (isDefinedAndNotNull(payload.user_phone_number)) {
-        payload.user_phone_number = sha256(
-          payload.user_phone_number
-        ).toString();
-      }
+      payload = {
+        user_hashed_email: userEmail ? sha256(userEmail).toString() : "",
+        user_hashed_phone_number: userPhoneNumber
+          ? sha256(userPhoneNumber).toString()
+          : "",
+      };
+    } else {
+      payload = {
+        user_email: userEmail,
+        user_phone_number: userPhoneNumber,
+      };
     }
```

Comment:
We can remove these checks if `userEmail` and `UserPhoneNumber` is not getting populated from the cookie. If the `removeUndefinedAndNullValues` is handling these case in line no. 96 then no need to change anything.

Reply:
Which checks we have to remove?? @ItsSudip

Reply:
We are populating user_hashed_email from userEmail what if userEmail is not populated from cookie? Then the payload will look like
```
{
     user_hashed_email: "",
     user_hsahed_phone_number:""
}
```

Reply:
```
{
     user_hashed_email: "",
     user_hsahed_phone_number:""
}
```
Yeah, if it's empty then it gets filtered by `removeUndefinedAndNullValues` functions. So, which check to remove??

Reply:
Ideally we shouldn't put empty string to these fields as far as I remember these fields are non mandatory. So even if the email, phoneNumber is not in the cookie we should init without these parameters

Reply:
Yeah, it's updated will raise the PR after checking in local for dedupKey setting from the webapp.